doc/specs/portable-mode-spec.md

```diff
@@ -0,0 +1,72 @@
+---
+author: Dustin L. Howett @DHowett
+created on: 2023-03-22
+last updated: 2023-03-22
+issue id: none
+---
+
+# Windows Terminal "Portable" Mode
+
+## Abstract
+
+Since we are planning on officially supporting unpackaged execution, I propose a special mode where Terminal stores its
+settings in a `settings` folder next to `WindowsTerminal.exe`.
+
+## Inspiration
+
+- [PortableApps](https://portableapps.com)
+- "Embeddable" Python, which relies on the deployment of a specific file to the Python root
+
+## Solution Design
+
+- _If running without package identity,_ `CascadiaSettings` will look for the presence of a file called `.portable` next
+  to `Microsoft.Terminal.Settings.Model.dll`.
+- If that file is present, it will change the settings and state paths to be rooted in a subfolder named `settings` next
+  to `Microsoft.Terminal.Settings.Model.dll`.
+
+Right now, _the only thing_ that makes Terminal not work in a "portable" manner is that it saves settings to
+`%LOCALAPPDATA%`.
+
+## UI/UX Design
+
+_No UI/UX impact is expected._
+
+## Capabilities
+
+- Distributors could ship a self-contained and preconfigured Terminal installation.
+- Users could archive fully-working preconfigured versions of Terminal.
+- Developers (such as those on the team) could easily test multiple versions of Terminal without worrying about global
+  settings pollution.
+
+### Accessibility
+
+_No change is expected._
+
+### Security
+
+_No change is expected._
+
+### Reliability
+
+More code always bears a risk.
+
+### Compatibility
+
+This is a net new feature, and it does not break any existing features. A distributor (or a user) can opt in (or out) by
+adding (or removing) the `.portable` file.
+
+### Performance, Power, and Efficiency
+
+_No change is expected._
+
+## Potential Issues

```

Comment:
How does this work with...
- dynamic profiles? Is it going to be an issue if a profile is added, but is invalid on another machine?
- fragment extensions? (basically the same question/answer as above, but figured I'd cover my bases)
- `"firstWindowPreference": "persistedWindowLayout"`? Specifically if we're pointing to a profile that doesn't exist anymore.
- moving between a Windows 10 and Windows 11 machine? (I think this one you said you fixed, but I want to double check)
- any settings that are more "machine specific" rather than "terminal specific" (i.e. `"experimental.rendering.forceFullRepaint"` and `"experimental.rendering.software"`)
- the shell extension (and running `wt.exe` from Run)? Do we run the portable mode version?

I think most of these scenarios are handled just fine, I just want to double check. They _might_ be worth mentioning in the spec too.

Reply:
- dynamic profiles are generated as normal; if the portable install is moved to somewhere where they don't apply, they will disappear
- fragment extensions are just another dynamic profile source; see above
- same thing that happens if you get rid of a dynamic profile in normal Terminal
- this is the same as upgrading a machine from windows 10 to 11 after installing Terminal
- "machine specific" settings still apply to the installation; we do not have (and should not have) a separate store for machine specific settings
- there is no shell extension registration in portable mode, but if we do it it is already set up to run wt from the same folder as the shell extension
- state is saved alongside settings, alwaus.